### PR TITLE
Add proxyID parameter to edsz

### PR DESCRIPTION
**Pull request by howardjohn**
Right now edsz isn't very useful since its using the legacy eds output
that isn't even accurate often. We are also doing a lot of processing
after the code we call, so it doesn't have info like locality lb, etc.

This adds both of those.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
Why is `proxyID` just for  edsz?  We donot get proxy specific listeners/route/clusters/endpoints from pilot.

**Comment by howardjohn** [comment]
the proxyID concept is not completely new, we use it in the config_dump endpoint as well.

the important difference is that EDS response is different for each proxy, so before it was impossible to debug things like locality lb, and the new transport socket match field. I added this because this was the only way for me to debug an issue I was having, so I think it's useful

**Comment by hzxuzhonghu** [comment]
But there is a possibility that this proxy is not connected to the current pilot if we have multi instances.

**Comment by howardjohn** [comment]
That is a good point... I copied the code from config_duml which istioctl uses, so I wonder how that works. Maybe istioctl queries all pilots?

**Comment by hzxuzhonghu** [comment]
`istioctl pc xxx `calls the kube-apiserver, and exec into the pod

**Comment by howardjohn** [comment]
There is also proxy-status that calls pilot config dump and compares to proxy config dump

**Comment by hzxuzhonghu** [comment]
Interesting, also did not consider multi pilot

**Comment by howardjohn** [comment]
@hzxuzhonghu @rshriram I don't think we can really fix the multi-pilot scenario, but this is still useful for debugging. Do you think its worth merging as is?

**Comment by hzxuzhonghu** [comment]
With explicit note, I think it makes sense.

**Comment by howardjohn** [comment]
@hzxuzhonghu sgtm, I added a note that it may be connected to another replica

File: pilot/pkg/proxy/envoy/v2/debug.go [hunk 1]
```diff
@@ -749,11 +749,48 @@ func (s *DiscoveryServer) edsz(w http.ResponseWriter, req *http.Request) {
...
+			}
+		}
+		con = connections[mostRecent]
+	}
 
 	edsClusterMutex.RLock()
 	defer edsClusterMutex.RUnlock()
 	comma := false
-	if len(edsClusters) > 0 {
+	if con != nil {
```
Comment (hzxuzhonghu) [inline pilot/pkg/proxy/envoy/v2/debug.go]: Isn't this condition always true?

Comment (hzxuzhonghu) [reply pilot/pkg/proxy/envoy/v2/debug.go]: ignore, i am wrong

Comment (howardjohn) [reply pilot/pkg/proxy/envoy/v2/debug.go]: no, it's only set if they pass proxyId

**Pull request merged**